src/transformers/models/whisper/configuration_whisper.py

```diff
@@ -0,0 +1,205 @@
+            features.
+        input_channels (`int`, *optional*, defaults to 1):
+            An integer specifying number of input channels of the input feature vector.
+
+    Example:
+
+    ```python
+    >>> from transformers import WhisperModel, WhisperConfig
+
+    >>> # Initializing a Whisper tiny style configuration
+    >>> configuration = WhisperConfig()
+
+    >>> # Initializing a model from the tiny style configuration
+    >>> model = WhisperModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "whisper"
+    keys_to_ignore_at_inference = ["past_key_values"]
+    attribute_map = {"num_attention_heads": "encoder_attention_heads", "hidden_size": "d_model"}
+
+    def __init__(
+        self,
+        vocab_size=51865,
+        feature_size=1,
+        num_mel_bins=80,
+        encoder_layers=6,
+        encoder_attention_heads=4,
+        decoder_layers=6,
+        decoder_attention_heads=4,
+        decoder_ffn_dim=1536,
+        encoder_ffn_dim=1536,
+        encoder_layerdrop=0.0,
+        decoder_layerdrop=0.0,
+        decoder_start_token_id=50258,
+        use_cache=True,
+        is_encoder_decoder=True,
+        activation_function="gelu",
+        d_model=256,
+        dropout=0.0,
+        attention_dropout=0.0,
+        activation_dropout=0.0,
+        init_std=0.02,
+        scale_embedding=False,
+        max_source_positions=1500,
+        max_target_positions=448,
+        pad_token_id=0,
+        bos_token_id=50257,
+        eos_token_id=50257,
+        input_channels=1,
+        tie_word_embeddings=True,
+        **kwargs
+    ):
+        self.vocab_size = vocab_size
+        self.num_mel_bins = num_mel_bins
+        self.d_model = d_model
+        self.encoder_layers = encoder_layers
+        self.encoder_attention_heads = encoder_attention_heads
+        self.decoder_layers = decoder_layers
+        self.decoder_attention_heads = decoder_attention_heads
+        self.decoder_ffn_dim = decoder_ffn_dim
+        self.encoder_ffn_dim = encoder_ffn_dim
+        self.dropout = dropout
+        self.attention_dropout = attention_dropout
+        self.activation_dropout = activation_dropout
+        self.activation_function = activation_function
+        self.init_std = init_std
+        self.encoder_layerdrop = encoder_layerdrop
+        self.decoder_layerdrop = decoder_layerdrop
+        self.use_cache = use_cache
+        self.num_hidden_layers = encoder_layers
+        self.scale_embedding = scale_embedding  # scale factor will be sqrt(d_model) if True
+        self.tie_word_embeddings = tie_word_embeddings
+        self.input_channels = input_channels
+        self.max_source_positions = max_source_positions
+        self.max_target_positions = max_target_positions
+        self.feature_size = feature_size
+        self.non_speech_tokens = NON_SPEECH_TOKENS
```

Comment:
This depends on whether the model is multi-lingual or not. IMO two options:

Option 1 (the cleaner):
- Just let the user input the speech tokens and let them default to `None`

Option 2 (keeps the config easier to read but not as clean):
- Add a flag to decide between multi-lingual and/or not

=> would prefer Option 1